### PR TITLE
Experimental CI test for private gce clusters.

### PR DESCRIPTION
**Pull request by mm4tt**
Ref. https://github.com/kubernetes/kubernetes/issues/76374

### CODE REVIEW

File: config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml [hunk 1]
```diff
@@ -540,3 +540,32 @@ periodics:
...
+          - --timeout=150
+          - --bare
+          - --scenario=kubernetes_e2e
+          - --
+          - --env=GCE_PRIVATE_CLUSTER=true
+          - --extract=ci/latest
+          - --gcp-master-image=gci
+          - --gcp-node-image=gci
+          - --gcp-nodes=10
+          - --gcp-project-type=scalability-project
```
Comment (wojtek-t) [inline config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml]: Why do we need scalability project?

File: config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml [hunk 2]
```diff
@@ -540,3 +540,32 @@ periodics:
...
+        args:
+          - --timeout=150
+          - --bare
+          - --scenario=kubernetes_e2e
+          - --
+          - --env=GCE_PRIVATE_CLUSTER=true
+          - --extract=ci/latest
+          - --gcp-master-image=gci
+          - --gcp-node-image=gci
+          - --gcp-nodes=10
```
Comment (wojtek-t) [inline config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml]: Do we really need 10 nodes?

File: testgrid/config.yaml [hunk 3]
```diff
@@ -6830,6 +6832,8 @@ dashboards:
     test_group_name: ci-kubernetes-e2e-gce-scale-correctness
   - name: gce-scale-performance
     test_group_name: ci-kubernetes-e2e-gce-scale-performance
+  - name: gce-private-cluster-correctness-experimental
```
Comment (wojtek-t) [inline testgrid/config.yaml]: I would feel more comfortable if we would introduce a dedicated tab for experimental jobs - I  think we start to have some and more potentially may come, so maybe it's time to do that.

Comment (mm4tt) [reply testgrid/config.yaml]: Done.

File: config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml (hunk 2 above)
Comment (mm4tt) [reply config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml]: Changed to 5.

File: config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml (hunk 1 above)
Comment (mm4tt) [reply config/jobs/kubernetes/sig-scalability/sig-scalability-periodic-jobs.yaml]: Removed.

**Pull request merged**